### PR TITLE
Private-transaction-families

### PR DESCRIPTION
**Pull request by sgurary**
As said, Private-transaction-families system provides a mechanism for privacy over Hyperledger Sawtooth by enforcing a policy of access control to the ledger and sensitive state content based on the client’s role trying to access the data.

This has over updated request details at labs/Private-transaction-families.md

### CODE REVIEW

**Comment by tkuhrt** [comment]
@sgurary : This PR has two proposals:
1. Private-transaction-families.md - Can you rename this to sawtooth-secure-transaction-families.md?
2. sawtooth-secure-transaction-families.md - Remove the old file from your fork. If I am not mistaken this is the one that we incorrectly created the last go around.

@dcmiddle : Can you review this as a sponsor?

File: labs/Private-transaction-families.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The system is based on a generic distributed ledger infrastructure (Hyperledger Sawtooth) and an Intel® SGX (Software Guard Extensions) ‘hardened’ Transaction Processor which provides a mechanism to enable Hyperledger Sawtooth to contain private information that is both not publicly available but required to validate transactions. This solution supports encryption of information in transactions and blocks while allowing the ledger to validate the information in those transactions in all the nodes and allow them to reach a consensus on the current state of the ledger.
+
+Private-transaction-families provides benefits for both application developers seeking to define and implement privacy-preserving distributed ledgers, and for service providers seeking to provide blockchain services with access restriction to the ledger data.
+
+# Scope of Lab
+
+Private-transaction-families system provides a mechanism for privacy over Hyperledger Sawtooth by enforcing a policy of access control to the ledger and sensitive state content based on the client’s role trying to access the data.
```
Comment (tkuhrt) [inline labs/Private-transaction-families.md]: This section should include what you consider in scope for the lab.

File: labs/Private-transaction-families.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Lab Name
+
+Private-transaction-families
+
+# Short Description
+
+Private-transaction-families system provides a mechanism for privacy over Hyperledger Sawtooth by enforcing a policy of access control to the ledger and sensitive state content based on the client’s role trying to access the data.
+
+The system is based on a generic distributed ledger infrastructure (Hyperledger Sawtooth) and an Intel® SGX (Software Guard Extensions) ‘hardened’ Transaction Processor which provides a mechanism to enable Hyperledger Sawtooth to contain private information that is both not publicly available but required to validate transactions. This solution supports encryption of information in transactions and blocks while allowing the ledger to validate the information in those transactions in all the nodes and allow them to reach a consensus on the current state of the ledger.
```
Comment (tkuhrt) [inline labs/Private-transaction-families.md]: This section should only contain a short one-sentence description of the lab that we can use when creating the Github repository.

File: labs/Private-transaction-families.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Private-transaction-families provides benefits for both application developers seeking to define and implement privacy-preserving distributed ledgers, and for service providers seeking to provide blockchain services with access restriction to the ledger data.
+
+# Scope of Lab
+
+Private-transaction-families system provides a mechanism for privacy over Hyperledger Sawtooth by enforcing a policy of access control to the ledger and sensitive state content based on the client’s role trying to access the data.
+
+The system is based on a generic distributed ledger infrastructure (Hyperledger Sawtooth) and an Intel® SGX (Software Guard Extensions) ‘hardened’ Transaction Processor which provides a mechanism to enable Hyperledger Sawtooth to contain private information that is both not publicly available but required to validate transactions. This solution supports encryption of information in transactions and blocks while allowing the ledger to validate the information in those transactions in all the nodes and allow them to reach a consensus on the current state of the ledger.
+
+# Documentation
```
Comment (tkuhrt) [inline labs/Private-transaction-families.md]: Remove, as this is not part of the template and does not currently contain any information.

File: labs/Private-transaction-families.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+- Will be updated with links later -
+
+Setup Instructions for Private-transaction-families can be found in the SETUP documentation.
+For more information about how Private-transaction-families works, see the SPECIFICATION document.
+The USAGE document describes How to run the system.
+
+# Project Status
```
Comment (tkuhrt) [inline labs/Private-transaction-families.md]: Remove as this is not part of the template and does not currently contain any information. Also, in labs world, we do not use the word "project".

File: labs/Private-transaction-families.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+- https://github.com/yoni-wolf
+- https://github.com/ronenshemtov
+- https://github.com/naishai
+- https://github.com/esorski
+
+# Sponsor
+
+Dan Middleton (dan.middleton@intel.com)
+
+#License
```
Comment (tkuhrt) [inline labs/Private-transaction-families.md]: Remove this section as this is not part of the proposal template.

**Comment by tkuhrt** [comment]
Please update to reflect the [proposal template](https://github.com/hyperledger-labs/hyperledger-labs.github.io/blob/master/proposal-template.md). Sections in your proposal should match those specified in the template.

**Comment by dcmiddle** [comment]
Ok, so there has been a little confusion but I think we are clearing it up. This lab should be called "private-transaction-families"
The old file should be removed and no one should ever say the old name again :)

**Comment by tkuhrt** [comment]
I am good with private-transaction-families for a lab name. Here are the things that need to be done:

1. Rename Private-transaction-families.md to private-transaction-families.md (capitalization)
2. Delete sawtooth-secure-transaction-families.md
3. Update private-transaction-families.md to match the proposal template and update the Scope of Lab and short description.

**Comment by sgurary** [comment]
Hi, I fixed the issues. @tkuhrt

**Comment by dcmiddle** [comment]
So close!!
Your newest commits don't have sign-offs. 
I think what you may want to do is rebase and squash all these commits into 1 commit and add the sign-off at the same time to that commit. (you can deselect the merge commit too)
something like
```
#in your branch
git fetch upstream #or whatever name you use for the hyperledger remote.. maybe origin
git rebase -i upstream/master --signoff  #again might be origin rather than upstream for you
# brings up an editor window with the list of commits...
# all of your commits are selected by default
# you can probably drop the merge commit
# save and exit the editor.. rebasing happens. 
# your remaining single commit should now have a signoff and will be in sync with master
git push -f yourfork yourbranch  # force push
```

**Comment by sgurary** [comment]
@dcmiddle can't make it work. any idea?

**Pull request closed without merging**